### PR TITLE
Add Dockerfile for IPython notebook

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

File: ipython/run_jupyter.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+jupyter notebook --ip='*' --no-browser
```
Comment (cancan101) [inline ipython/run_jupyter.sh:3]: I think to be safe you want to `exec` this.

Comment (taion) [inline ipython/run_jupyter.sh:3]: Why's that?

**Comment by cancan101** [comment]
It moves the `jupyter` command to `pid 1` (which I beleive has some special semantics for signal handling). Also means only one process running in container rather than bash + jupyter.

w/o `exec`:

```
$ docker exec 2da15f24bbbe ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.2  17964  2880 ?        Ss+  02:25   0:00 bash /root/less.sh
root         8  0.0  0.1   6700  1644 ?        S+   02:25   0:00 less /root/f
```

with `exec`:

```
$ docker exec 015e79b13d9e  ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.1  0.1   6700  1652 ?        Ss+  02:26   0:00 less /root/f
```

where bash runs less.

**Comment by taion** [comment]
The TF equivalent does not have that behavior, though: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/run_jupyter.sh#L18.

**Pull request merged**